src/typemap.c

```diff
@@ -1127,58 +1123,50 @@ static jl_typemap_level_t *jl_method_convert_list_to_cache(
 
 static void jl_typemap_list_insert_(
         jl_typemap_t *map, jl_typemap_entry_t **pml, jl_value_t *parent,
-        jl_typemap_entry_t *newrec, const struct jl_typemap_info *tparams)
+        jl_typemap_entry_t *newrec)
 {
-    if (*pml == (void*)jl_nothing || newrec->isleafsig || (tparams && tparams->unsorted)) {
-        newrec->next = *pml;
-        jl_gc_wb(newrec, newrec->next);
-        jl_atomic_store_release(pml, newrec);
-        jl_gc_wb(parent, newrec);
-    }
-    else {
-        jl_typemap_list_insert_sorted(map, pml, parent, newrec, tparams);
-    }
+    newrec->next = *pml;
```

Comment:
I think you may still want to sort them on their lookup performance kind (isleafsig, then issimplesig, then remainder)
```suggestion
    jl_typemap_entry_t *l = *pml;
    while ((jl_value_t*)l != jl_nothing) {
        if (newrec->isleafsig || !l->isleafsig)
            if (newrec->issimplesig || !l->issimplesig)
                break;
        pml = &l->next;
        parent = (jl_value_t*)l;
        l = l->next;
     }
    newrec->next = l;
```